### ISSUE TITLE
Does not recognise git on windows

### ISSUE THREAD

**Issue by TTitcombe** [DocumaticAI/doculog#27]
**Describe the bug**
Doculog uses the python package `subprocess` to run git under the hood to collect commits.
This works fine for linux (i believe macOS as well), but fails with windows because subprocess cannot find.

We have introduced a function to get git from path, but that does not seem to work either (https://github.com/DocumaticAI/doculog/blob/master/doculog/git.py#L13).
We have tried setting shell=True, but that also doesn't seem to work 
Can someone please confirm the above?

**To Reproduce**
1. Be on a windows machine
2. Run `subprocess.check_output["git", "log"])
3. This will fail with a `FileNotFoundError`

**Expected behavior**
We should be able to get git logs on windows!

A fix to this issue will update `doculog/git.py` to work on windows and non-windows with the same functionality

**Screenshots**
None

**System:**

- OS: windows
- Doculog: all

**Additional context**
N/A

**Comment by TheQuizzer** [comment]
Open a Command Prompt. Run setx GCM_TRACE %UserProfile%\git. log ; this will cause the GCM to log to a file located at “%UserProfile%\git. log”.

**Comment by Rahuletto** [comment]
Install git from their site ☺️

**Comment by Rahuletto** [comment]
Install git from https://git-scm.com/

**Comment by TTitcombe** [comment]
@Rahuletto we have no control over how users have installed git. This problem exists with users who have working git, but subprocess cannot access it properly (I don't know the root cause of this)

**Comment by BasToTheMax** [comment]
I don't know how to code in python, so i will send a pseudocode in here:

You could locate the path of the `git.exe` file https://intellipaat.com/community/23035/where-is-git-exe-located

and then do something like this:
`exec_command("C:\Program Files (x86)\Git\Git.exe git commit)`

**Comment by gXLg** [comment]
Problem is, it is sometimes not git.exe, but git.cmd

**Comment by pro-gamer007** [comment]
You can probably find what it is thru the system path

**Comment by pgamerx** [comment]
I think that in windows, it is never installed as `git.exe`, it is installed as `git.cmd`.


> I don't know how to code in python, so i will send a pseudocode in here:
> 
> You could locate the path of the `git.exe` file https://intellipaat.com/community/23035/where-is-git-exe-located
> 
> and then do something like this: `exec_command("C:\Program Files (x86)\Git\Git.exe commit)`

**Comment by Lightninbolt986** [comment]
> C:\Program Files (x86)\Git\

After a bit of research, it seems it may be installed in C:\Program Files\Git too.
I also found that git may be in C:\Program Files\Git\cmd\git.exe

**Comment by L0RD-ZER0** [comment]
Here is a way I came up with:

```py
from subprocess import check_output 
from sys import stdout
from os.path import splitext


valid_ext = ['.exe', 'cmd', 'bat']
def filter_func(val: str) -> bool:
    if splitext(val)[1] in valid_ext:
        return True


where_output: list[str] = check_output(['where', 'git'])\
                                               .decode(stdout.encoding)\
                                               .split('\n')
# decode to decode bytes
# split just in case there are multiple files
# then filter to remove in case there's something which shouldn't be run

git_location = list(filter(filter_func, where_output))[0].strip()


print(git_location)
```

So maybe try using something this?

**Comment by Wumpuspro** [comment]
Locate the path of the git.exe to this file https://intellipaat.com/community/23035/where-is-git-exe-located

**Comment by WhiteClueGaming** [comment]
Instead of subprocess.check_output["git", "log"]) use subprocess.check_output["git", "log"], stdin=date_proc.stdout, stdout=subprocess.PIPE)

**Comment by Assassin-1234** [comment]
please create a pull request fixing the issue mentioned above

**Comment by Assassin-1234** [comment]
@L0RD-ZER0 can I have your discord tag?

**Comment by L0RD-ZER0** [comment]
> @L0RD-ZER0 can I have your discord tag?

You can use`The One#9690` to get in touch.

**Comment by pgamerx** [comment]
Wasn't this fixed in https://github.com/DocumaticAI/doculog/commit/fa6018bfabe400a0355f84e7cb84f4e746aac12c

**Comment by TTitcombe** [comment]
@pgamerx That's correct. Thanks, closing now.

Fixed by #29
